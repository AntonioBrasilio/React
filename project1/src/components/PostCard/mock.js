export const PostCardPropsMock = {
    title: "mock-title",
    body: "mock-body",
    id: 1,
    cover: "img/mock-img.png",
};
