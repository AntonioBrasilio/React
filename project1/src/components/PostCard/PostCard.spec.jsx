import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { PostCardPropsMock } from "./mock";

const props = PostCardPropsMock;

describe("<PostCard />", () => {
    it("should render PostCard correctly in the screen", () => {
        expect.assertions(3);

        render(<PostCard {...props} />);

        expect(screen.getByRole("img", { name: props.title })).toHaveAttribute(
            "src",
            props.cover
        );

        expect(
            screen.getByRole("heading", { name: props.title })
        ).toBeInTheDocument();

        expect(screen.getByText(props.body)).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect.assertions(1);

        render(<PostCard {...props} />);
        expect(screen.getByTestId('post-card-test-id')).toMatchSnapshot();
    });
});
