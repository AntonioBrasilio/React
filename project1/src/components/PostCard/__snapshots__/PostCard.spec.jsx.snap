// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
  data-testid="post-card-test-id"
>
  <img
    alt="mock-title"
    src="img/mock-img.png"
  />
  <div
    class="post-content"
    id="1"
  >
    <h2>
      mock-title
    </h2>
    <p>
      mock-body
    </p>
  </div>
</div>
`;
